gobblin-runtime/src/main/java/org/apache/gobblin/quality/DataQualityEvaluator.java

```diff
@@ -0,0 +1,221 @@
+                } else if (DataQualityStatus.FAILED == taskDataQuality){
+                    failedFilesCount++;
+                    jobDataQualityStatus = DataQualityStatus.FAILED;
+                } else {
+                    log.warn("Unexpected data quality status: " + taskDataQuality + " for task: " + taskState.getTaskId());
+                }
+            } else {
+                // Handle files without data quality evaluation
+                nonEvaluatedFilesCount++;
+                log.warn("No data quality evaluation for task: " + taskState.getTaskId());
+            }
+        }
+
+        // Log summary of evaluation
+        log.info("Data quality evaluation summary - Total: {}, Passed: {}, Failed: {}, Not Evaluated: {}",
+            totalFiles, passedFilesCount, failedFilesCount, nonEvaluatedFilesCount);
+        return new DataQualityEvaluationResult(jobDataQualityStatus, totalFiles, passedFilesCount, failedFilesCount, nonEvaluatedFilesCount);
+    }
+
+
+    private static void emitMetrics(JobState jobState, final DataQualityStatus jobDataQuality, final int totalFiles,
+            final int passedFilesCount, final int failedFilesCount, final int nonEvaluatedFilesCount, final String datasetUrn) {
+        try {
+            // Check if OpenTelemetry is enabled
+            boolean otelEnabled = jobState.getPropAsBoolean(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_ENABLED,
+                ConfigurationKeys.DEFAULT_METRICS_REPORTING_OPENTELEMETRY_ENABLED);
+
+            if (!otelEnabled) {
+                log.info("OpenTelemetry metrics disabled, skipping metrics emission");
+                return;
+            }
+
+            OpenTelemetryMetricsBase otelMetrics = OpenTelemetryMetrics.getInstance(jobState);
+            log.info("OpenTelemetry instance obtained: {}", otelMetrics != null);
+
+            if (otelMetrics != null) {
+                Attributes tags = getTagsForDataQualityMetrics(jobState, datasetUrn);
+                log.info("Tags for data quality metrics: " + tags.toString());
+                // Emit data quality status (1 for PASSED, 0 for FAILED)
+                log.info("Data quality status for this job is " + jobDataQuality);
+                if (jobDataQuality == DataQualityStatus.PASSED) {
+                    log.info("Data quality passed for job: {}", jobState.getJobName());
+                    otelMetrics.getMeter(GAAS_OBSERVABILITY_METRICS_GROUPNAME)
+                        .counterBuilder(ServiceMetricNames.DATA_QUALITY_JOB_SUCCESS_COUNT)
+                        .setDescription("Number of jobs that passed data quality")
+                        .build()
+                        .add(1, tags);
+                } else {
+                    log.info("Data quality failed for job: {}", jobState.getJobName());
+                    otelMetrics.getMeter(GAAS_OBSERVABILITY_METRICS_GROUPNAME)
+                        .counterBuilder(ServiceMetricNames.DATA_QUALITY_JOB_FAILURE_COUNT)
```

Comment:
since if/else differs in metric name only, we can consider determining the metric name first and remove this if/else block using below:
```
String jobMetricName = (jobDataQuality == DataQualityStatus.PASSED)
        ? ServiceMetricNames.DATA_QUALITY_JOB_SUCCESS_COUNT
        : ServiceMetricNames.DATA_QUALITY_JOB_FAILURE_COUNT;
```